html_versions.py

```diff
@@ -0,0 +1,239 @@
+"""
+Inspired by Tom's /dls/cs-www/reports/scripts/list_ioc_support_modules.py
+"""
+
+import pkg_resources
+pkg_resources.require('jinja2')
+import jinja2
+
+from convert import configuration
+from convert import coordinates
+from convert import dependency
+from convert import launcher
+from convert import utils
+
+import os
+import subprocess
+import ConfigParser
+import logging as log
+LOG_FORMAT = '%(levelname)s:  %(message)s'
+LOG_LEVEL = log.WARNING
+log.basicConfig(format=LOG_FORMAT, level=LOG_LEVEL)
+
+SUPPORT = '/dls_sw/prod/R3.14.12.3/support'
+IOC = '/dls_sw/prod/R3.14.12.3/ioc'
+RESOURCES_DIR = 'res'
+HTML_TEMPLATE = 'template.html'
+REPORT = 'deps.html'
+CFG_IOC_CMD = ["configure-ioc", "l"]
+
+
+class ModDetails(object):
+    # CSS classes
+    OK = 'ok'
+    OUT_OF_DATE = 'out-of-date'
+    CONFIGURED = 'configured'
+
+    def __init__(self, name, requested, latest_release, config_version):
+        self.name = name
```

Comment:
Constructor is clumsy; you use it twice both times setting additional state in the following lines. Would creating two constructors tailored to the two use-cases (dependency or top-level module) be simpler to use?